googleapiclient/http.py

```diff
@@ -56,6 +56,10 @@
 from googleapiclient.model import JsonModel
 from oauth2client import util
 
+try:
```

Comment:
Fair enough. Is there any worry about catching socket errors that aren't timeouterrors that we should do different stuff with?